### PR TITLE
[SDP-1089] Add rate-limiters both in the application and the k8s deployment

### PR DESCRIPTION
**Pull request by marcelosalloum**
### What

Implement in-memory rate-limiting in the server, sensitive to the pair {IP, endpoint}. It's configured to a max of 10 requests every 20 seconds, which translates to a maximum of 30 requests per minute per IP, segregated per endpoint.

Also, add rate-limit to the k8s Ingress manifest.

### Why

For increased security against bots.

#### PR Structure

* [x] This PR has a reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR title and description are clear enough for anyone to review it.
* [x] This PR does not mix refactoring changes with feature changes (split into two PRs otherwise).

#### Thoroughness

* [x] This PR adds tests for the new functionality or fixes.
* [x] This PR contains the link to the Jira ticket it addresses.

#### Configs and Secrets

* [x] No new CONFIG variables are required -OR- the new required ones were added to the helmchart's [`values.yaml`] file.
* [x] No new CONFIG variables are required -OR- the new required ones were added to the deployments ([`pr-preview`], [`dev`], [`demo`], `prd`).
* [x] No new SECRETS variables are required -OR- the new required ones were mentioned in the helmchart's [`values.yaml`] file.
* [x] No new SECRETS variables are required -OR- the new required ones were added to the deployments ([`pr-preview secrets`], [`dev secrets`], [`demo secrets`], `prd secrets`).

#### Release

* [x] This is not a breaking change.
* [x] **This is ready for production.**. If your PR is not ready for production, please consider opening additional complementary PRs using this one as the base. Only merge this into `develop` or `main` after it's ready for production!

#### Deployment

* [x] Does the deployment work after merging?

[`values.yaml`]: ../helmchart/sdp/values.yaml
[`pr-preview`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/stellar-disbursement-platform/backend-helm-values
[`dev`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/backend-helm-values
[`demo`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-backend-helm-values
[`pr-preview secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/externalsecrets-common-previews.yaml#L241-L346
[`dev secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/stellar-disbursement-platform-externalsecrets.yaml
[`demo secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-sdp-externalsecrets.yaml

### CODE REVIEW

File: internal/serve/serve.go [hunk 1]
```diff
@@ -207,17 +208,25 @@ func Serve(opts ServeOptions, httpServer HTTPServerInterface) error {
...
+)
+
 func handleHTTP(o ServeOptions) *chi.Mux {
 	mux := chi.NewMux()
 
 	// Middleware
 	mux.Use(middleware.CorsMiddleware(o.CorsAllowedOrigins))
 	mux.Use(chimiddleware.RequestID)
 	mux.Use(chimiddleware.RealIP)
+	mux.Use(httprate.LimitAll(rateLimitPer20Seconds, rateLimitWindow))
```
Comment (marwen-abid) [inline internal/serve/serve.go:223]: Is this 90 rpm for all IPs or 90rpm per IP ?

**Review by marwen-abid — APPROVED** [review]
LGTM

File: internal/serve/serve.go (hunk 1 above)
Comment (CaioTeixeira95) [reply internal/serve/serve.go:223]: Looking at the [source code](https://github.com/go-chi/httprate/blob/master/httprate.go#L17-L19) looks like it's for all IPs. I'd suggest we use the following configuration:

```suggestion
	mux.Use(httprate.Limit(
		rateLimitPer20Seconds,
		rateLimitWindow,
		httprate.WithKeyFuncs(httprate.KeyByIP, httprate.KeyByEndpoint),
	))
```

**Review by CaioTeixeira95 — APPROVED** [review]

File: internal/serve/serve.go (hunk 1 above)
Comment (marcelosalloum) [reply internal/serve/serve.go:223]: Good catch, TY!

Comment (marcelosalloum) [reply internal/serve/serve.go]: I followed Caio's suggestion in https://github.com/stellar/stellar-disbursement-platform-backend/pull/195/commits/303a904b5738efc729ad107d411ca544f6a504ea, and updated the limit per window.

The new limit is 10 requests every 20 seconds with the pair {IP, endpoint}. Which translates to a maximum of 30 requests per minute per IP, segregated per endpoint.

**Pull request merged**